### PR TITLE
allow {} in ldap.search-filter

### PR DESCRIPTION
**Pull request by fceller**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.7

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [x] Docs PR: https://github.com/arangodb/docs/pull/611
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/636
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by fceller** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13714/

**Comment by fceller** [comment]
In "ldap.search-filter" the pattern "{USER}" will be replaced by the username

File: CHANGELOG [hunk 1]
```diff
@@ -1,39 +1,41 @@
 devel
 -----
 
+* Allow {} paceholder string also in `--ldap.search-filter`.
```
Comment (KVS85) [inline CHANGELOG:4]: * Allow {USER} paceholder string also in `--ldap.search-filter`.

**Review by KVS85 — COMMENTED** [review]
LGTM.

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13716/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13721/

**Review by joerg84 — APPROVED** [review]

**Review by jsteemann — APPROVED** [review]

**Comment by KVS85** [comment]
Tests blue except unrelated failure.

**Pull request merged**